### PR TITLE
[MIG] crm_claim: Migration to v12

### PR DESCRIPTION
**Pull request by cristinamartinrod**
cc @Tecnativa

### CODE REVIEW

File: crm_claim/models/__init__.py [hunk 1]
```diff
@@ -1,8 +1,7 @@
-# -*- coding: utf-8 -*-
-# Copyright 2015-2017 Odoo S.A.
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
 
 from . import crm_claim
 from . import crm_claim_category
 from . import crm_claim_stage
 from . import res_partner
+from . import res_request
```
Comment (sergio-teruel) [inline crm_claim/models/__init__.py]: Not needed

File: crm_claim/__manifest__.py [hunk 2]
```diff
@@ -22,6 +21,7 @@
         'views/crm_claim_stage_views.xml',
         'views/res_partner_views.xml',
         'views/crm_claim_menu.xml',
+        'views/res_request_view.xml',
```
Comment (sergio-teruel) [inline crm_claim/__manifest__.py]: Not needed.

File: crm_claim/models/crm_claim.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
-# -*- coding: utf-8 -*-
 # Copyright 2015-2017 Odoo S.A.
-# Copyright 2017 Vicent Cubells <vicent.cubells@tecnativa.com>
+# Copyright 2017 Tecnativa - Vicent Cubells
+# Copyright 2018 Tecnativa - Cristina Martin R.
```
Comment (sergio-teruel) [inline crm_claim/models/crm_claim.py]: If you migrate a module you has not the copyright, only as a contributor

File: crm_claim/models/crm_claim.py [hunk 4]
```diff
@@ -26,6 +24,10 @@ def _get_default_stage_id(self):
     def _get_default_team(self):
         return self.env['crm.team']._get_default_team_id()
 
+    @api.model
+    def referenceable_models(self):
+        return [(link.model, link.name) for link in self.env['res.request.link'].search([])]
+
```
Comment (sergio-teruel) [inline crm_claim/models/crm_claim.py:40]: Remove this method, add directly as a tuple ('model', 'name') in selection field attribute.

File: crm_claim/models/crm_claim.py [hunk 5]
```diff
@@ -56,7 +58,7 @@ def _get_default_team(self):
         detault=fields.Datetime.now,
     )
     model_ref_id = fields.Reference(
-        selection=odoo.addons.base.res.res_request.referenceable_models,
+        selection='referenceable_models',
```
Comment (sergio-teruel) [inline crm_claim/models/crm_claim.py]: Add directly selection=[('res.partner', 'Partner'))]... mhuu... if model not exists when select the record we will have a key error, so I think that if you want add a model that other module add it you must be depend of this module...
@pedrobaeza We do you can with this model...??

File: crm_claim/models/res_request.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+from odoo import api, fields, models
+
+
+class ResRequestLink(models.Model):
+    _name = 'res.request.link'
+    _order = 'priority'
+
+    name = fields.Char(required=True, translate=True, index=True)
+    model = fields.Char(required=True, index=True)
+    priority = fields.Integer(default=5)
```
Comment (sergio-teruel) [inline crm_claim/models/res_request.py]: This model has not been in this module...

File: crm_claim/report/crm_claim_report_view.xml [hunk 7]
```diff
@@ -36,15 +36,15 @@
         <field name="name">crm.claim.report.select</field>
         <field name="model">crm.claim.report</field>
         <field name="arch" type="xml">
-            <search string="Search">
+            <form>
```
Comment (sergio-teruel) [inline crm_claim/report/crm_claim_report_view.xml]: This is a search view... so your deletion is not correct

File: crm_claim/report/crm_claim_report_view.xml [hunk 8]
```diff
@@ -54,7 +54,7 @@
                     <field name="date_closed" />
                     <field name="date_deadline" />
                 </group>
-                <group  expand="1" string="Group By">
+                <group expand="1" string="Group By">
                     <filter string="Salesperson" name="Salesperson" context="{'group_by':'user_id'}" />
                     <filter string="Partner" name="partner" context="{'group_by':'partner_id'}" />
                     <filter string="Stage" context="{'group_by':'stage_id'}" />
```
Comment (sergio-teruel) [inline crm_claim/report/crm_claim_report_view.xml]: add attribute name for all filters

File: crm_claim/security/ir.model.access.csv [hunk 9]
```diff
@@ -2,6 +2,7 @@ id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
 access_crm_claim_manager,crm.claim.manager,model_crm_claim,sales_team.group_sale_manager,1,1,1,1
 access_crm_claim_user,crm.claim.user,model_crm_claim,sales_team.group_sale_salesman,1,1,1,0
 access_crm_claim_stage_user,crm.claim.stage.user,model_crm_claim_stage,sales_team.group_sale_salesman,1,1,1,1
+access_res_request_link_user,res.request.link.user,model_res_request_link,sales_team.group_sale_salesman,1,1,1,1
```
Comment (sergio-teruel) [inline crm_claim/security/ir.model.access.csv]: In other module??

File: crm_claim/views/crm_claim_views.xml [hunk 10]
```diff
@@ -103,27 +103,29 @@
...
-                    <filter string="Partner" domain="[]" help="Partner" context="{'group_by':'partner_id'}"/>
-                    <filter string="Responsible" domain="[]" help="Responsible User" context="{'group_by':'user_id'}"/>
-                    <filter string="Stage" domain="[]" context="{'group_by':'stage_id'}"/>
-                    <filter string="Type" domain="[]" context="{'group_by':'categ_id'}"/>
-                    <filter string="Claim Month" domain="[]" help="Claim Date by Month" context="{'group_by':'date'}"/>
-                    <filter string="Deadline" domain="[]" context="{'group_by':'date_deadline'}"/>
-                    <filter string="Closure" domain="[]" help="Date Closed" context="{'group_by':'date_closed'}"  groups="base.group_no_one" invisible="1"/>
-                </group>
-            </search>
+            <form>
```
Comment (sergio-teruel) [inline crm_claim/views/crm_claim_views.xml]: this is a search view

File: crm_claim/views/crm_claim_views.xml [hunk 11]
```diff
@@ -103,27 +103,29 @@
...
+                    <field name="partner_id" filter_domain="[('partner_id','child_of',self)]"/>
+                    <field name="user_id"/>
+                    <group expand="0" name="claims" string="Group By">
+                        <filter string="Partner" domain="[]" help="Partner" context="{'group_by':'partner_id'}"/>
+                        <filter string="Responsible" domain="[]" help="Responsible User" context="{'group_by':'user_id'}"/>
+                        <filter string="Stage" domain="[]" context="{'group_by':'stage_id'}"/>
+                        <filter string="Type" domain="[]" context="{'group_by':'categ_id'}"/>
+                        <filter string="Claim Month" domain="[]" help="Claim Date by Month" context="{'group_by':'date'}"/>
+                        <filter string="Deadline" domain="[]" context="{'group_by':'date_deadline'}"/>
+                        <filter string="Closure" domain="[]" help="Date Closed" context="{'group_by':'date_closed'}"  groups="base.group_no_one" invisible="1"/>
```
Comment (sergio-teruel) [inline crm_claim/views/crm_claim_views.xml]: name for filter

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

File: crm_claim/models/crm_claim.py [hunk 12]
```diff
@@ -26,6 +23,21 @@ def _get_default_stage_id(self):
...
+            ('sale.order', _('Sale Order')),
+            ('account.invoice', _('Invoices')),
+            ('purchase.order', _('Purchase Order')),
+            ('event.registration', _('Event Registration')),
+            ('account.invoice', _('Invoice')),
+            ('sale.order.line', _('Sale Order Line')),
+            ('purchase.order.line', _('Purchase Order Line')),
+            ('hr.applicant', _('HR Applicant')),
+            ('product.product', _('Products'))]
+        return model_filter
```
Comment (sergio-teruel) [inline crm_claim/models/crm_claim.py:39]: Do 
return list(filter( x: x[0] in self.env.registry.models, model_list))

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

File: crm_claim/models/crm_claim.py (hunk 12 above)
Comment (pedrobaeza) [reply crm_claim/models/crm_claim.py:39]: And don't put the translatable name. You already have it on the model. You can do:
```python
APPLICABLE_MODELS = [
    'res.partner',
    'sale.order',
    'account.invoice',
    'purchase.order',
    'event.registration',
    'account.invoice',
    'sale.order.line',
    'purchase.order.line',
    'hr.applicant',
    'product.product',
]

def _selection_model(self):
    return [(x, self.env[x]._description) for x in APPLICABLE_MODELS if x in self.env]
```

The only doubt right now is about translating that description. I think `_(self.env[x]._description)` is not going to work, so maybe you need to unfold a bit that and access to model name in ir.model.

**Pull request closed without merging**